### ISSUE TITLE
Proof of Void with equalities in Type

### ISSUE THREAD

**Issue by anton-trunov** [idris-lang/Idris-dev#4282]
# Steps to Reproduce
```
total
bad1 : (f : Type -> Type) -> (x, y : Type) -> f x = f y -> Void
bad1 _ _ _ Refl impossible

total
ohoh : Void
ohoh = bad1 id Unit Unit Refl

-- of course, these shouldn't be provable as well:
total
bad2 : (f, g : Type -> Type) -> (x : Type) -> f x = g x -> Void
bad2 _ _ _ Refl impossible

total
bad3 : (f, g : Type -> Type) -> (x, y : Type) -> f x = g y -> Void
bad3 _ _ _ _ Refl impossible
```
# Expected Behavior

`bad1`, `bad2`, `bad3` are not provable

# Observed Behavior

`bad1`, `bad2`, `bad3` are provable

# Notes

Tested with Idris 1.1.1-git:05634d71 (macOS 10.13.2 (17C88), High Sierra)

**Comment by ahmadsalim** [comment]
Thanks for reporting the issue!

It should not recognize these cases as impossible since `f x = f y` should get stuck and not fail.
Agda gives the following error:

```
I'm not sure if there should be a case for the constructor refl,
because I get stuck when trying to solve the following unification
problems (inferred index ≟ expected index):
  f x ≟ f y
when checking that the pattern refl has type f x ≡ f y
```

**Comment by anton-trunov** [comment]
Thanks for checking it with Agda. Coq also rejects this thing.

**Comment by melted** [comment]
elabClause for an impossible clause is

```idris
elabClause info opts (_, PClause fc fname lhs_in [] PImpossible [])
   = do let tcgen = Dictionary `elem` opts
        i <- get
        let lhs = addImpl [] i lhs_in
        b <- checkPossible info fc tcgen fname lhs_in
        case b of
            Just _ -> tclift $ tfail (At fc
                                 (Msg $ show lhs_in ++ " is a valid case"))
            Nothing -> do ptm <- mkPatTm lhs_in
                          logElab 5 $ "Elaborated impossible case " ++ showTmImpls lhs ++
                                      "\n" ++ show ptm
                          return (Left ptm, lhs)
```

problem is that checkPossible returns `Nothing`not because it is impossible but because typechecking fails:
```
Impossible case .\.\i4282.idr:3:6:When checking an application of Main.bad1:
        Type mismatch between
                f y = f y (Type of Refl)
        and
                f x = f y (Expected type)

        Specifically:
                Type mismatch between
                        f y
                and
                        f x
(False,False)
Elaborated impossible case Main.bad1 _
```
Either there should be a checkImpossible or checkPossible should throw if the error doesn't mean impossibilty.

**Comment by yanok** [comment]
Fixed by #4303
